pyvrp/cpp/search/OptionalInsert.cpp

```diff
@@ -0,0 +1,82 @@
+#include "OptionalInsert.h"
+#include "primitives.h"
+
+using pyvrp::search::OptionalInsert;
+
+void OptionalInsert::operator()(PerturbationContext const &context)
```

Comment:
There's lots of duplication here with `LocalSearch::insert`, I think. Can we somehow merge/abstract the relevant bits?

Reply:
(do we still have `repair::greedyRepair` in the ILS branch? Because that also somewhat duplicates this logic)

Reply:
I think `repair` can be completely removed with ILS.

Reply:
I made an attempt by introducing the `bestInsert` function at https://github.com/PyVRP/PyVRP/pull/877/commits/9e495cb7f33eb9e616565f7ab184da246fe49bc5. I'm not sure if this is better. We will only have two places where we use insertion (`LS::insert` and `InsertOptional`) so it's not that much duplication.